File: src/main/java/uk/edu/populationfitness/models/genes/performance/TimingStatistics.java
package uk.edu.populationfitness.models.genes.performance;

class TimingStatistics {

    private final String name;
    private double total;
    private int count;
    private long min;
    private long max;

    public TimingStatistics(String name){
        this.name = name;
        reset();
    }

    public void add(long value){
        count++;
        total += value;
        min = Math.min(value, min);
        max = Math.max(value, max);
    }

    private int count(){
        return this.count;
    }

    public long min(){
        return count == 0 ? 0 : this.min;
    }

    public long max(){
        return count == 0 ? 0 : this.max;
    }

    private long mean(){
        return count > 0 ? Math.round(total / count): 0;
    }

    public void reset()
    {
        total = 0;
        count = 0;
        min = Long.MAX_VALUE;
        max = Long.MIN_VALUE;
    }

    public void show(){
        System.out.print(name);
        if (count > 0){
            System.out.print(" Min=");
            System.out.print(min);
            System.out.print("(micros) Max=");
            System.out.print(max);
            System.out.print("(micros) Mean=");
            System.out.print(mean());
            System.out.print("(micros) Num=");
            System.out.print(count());
            System.out.print(" Tot=");
            System.out.print(Math.round(total) / 1000);
            System.out.println(" (millis)");
        }
        else
        {
            System.out.println(" None");
        }
    }
}

File: DotNet/PopulationFitness/PopulationFitness/Models/Genes/Performance/TimingStatistics.cs
using System;
using System.Diagnostics;

namespace PopulationFitness.Models.Genes.Performance
{
    class TimingStatistics
    {

        private readonly String _name;
        private double _total;
        private int _count;
        private long _max;
        private long _min;

        public TimingStatistics(String name)
        {
            this._name = name;
            Reset();
        }

        public void Add(long value)
        {
            _count++;
            _total += value;
            _min = Math.Min(value, _min);
            _max = Math.Max(value, _max);
        }

        public long Min
        {
            get
            {
                return _count == 0 ? 0 : _min;
            }
        }

        public long Max
        {
            get
            {
                return _count == 0 ? 0 : _max;
            }
        }

        private long Mean
        {
            get
            {
                return _count > 0 ? (long)Math.Round(_total / _count) : 0;
            }
        }

        public void Reset()
        {
            _total = 0;
            _count = 0;
            _min = long.MaxValue;
            _max = long.MinValue;
        }

        public void Show()
        {
            Debug.Write(_name);
            if (_count > 0)
            {
                Debug.Write(" Min=");
                Debug.Write(Min);
                Debug.Write("(micros) Max=");
                Debug.Write(Max);
                Debug.Write("(micros) Mean=");
                Debug.Write(Mean);
                Debug.Write("(micros) Num=");
                Debug.Write(_count);
                Debug.Write(" Tot=");
                Debug.Write(Math.Round(_total) / 1000);
                Debug.WriteLine(" (millis)");
            }
            else
            {
                Debug.WriteLine(" None");
            }
        }
    }
}
